### DISCUSSION TITLE
BBS on the Ret

### DISCUSSION THREAD

**Discussion by acehoss** [markqvist/Reticulum#231] [Show and tell]
With `r n s h` becoming more usable by the day, I thought it would be fun to open something up on Testnet that folks could connect to using it. Install `r n s h` and stop on down to play a classic game of minesweeper! (that's under External Programs > Games on the main menu)

![CleanShot 2023-02-19 at 15 06 34@2x](https://user-images.githubusercontent.com/5148966/219975252-3a0d7ea3-7d60-4dca-a169-1ca9ab486dfb.png)

If you've installed `nomadnet` (and are running at least Python 3.9 on macOS, Linux, Android, or WSL), you should hopefully not have too much trouble. `pip3 install rnsh`

After that, connecting is as simple as running `rnsh feda42292d882c5586e548d79b5af27b`

At this point I don't really have much of the proper BBS stuff set up. The main purpose is to test `r n s h` a little harder. It's still very fresh software and there are certainly bugs--but without testing, we won't find them! If you run into any issues--particularly issues that you can replicate multiple times with the same steps--let me know on the [`r n s h` GitHub page](https://github.com/acehoss/rnsh).

### UPDATE 2023-03-08
**`rnsh` v0.0.13 is now required** to connect to the BBS. Run `pip3 install -U rnsh` to update. As part of the update, the destination hash changed. The hash is unchanged,`feda42292d882c5586e548d79b5af27b `.

**Comment by 4c3e** [comment]
This is awesome! Nice work!

**Comment by Erethon** [comment]
Just want to say again congrats and good job on writing rnsh! I tested the BBS again over a LoRa connection using two RNodes and it provides a pretty authentic BBS experience due to the low speeds ^_^.

Here's a webm showing the BBS loading in real time 
[Peek 2023-02-20 16-30.webm](https://user-images.githubusercontent.com/1254842/220134979-755b297f-7007-49ae-839f-ec74cdd5761e.webm)

**Comment by markqvist** [comment]
This is really quite something :) Well done @acehoss! I've had a few sessions and everything worked flawlessly. It's the first time I've played minesweeper in ages :)

**Comment by faragher** [comment]
This is exactly what I hoped the concept would evolve into. The BBS was designed for exactly this kind of system, a plurality of severs with relatively swift bulk data synchronizations, all connecting to individual stations with a slow uplink.  

Amazing and thank you.

**Comment by acehoss** [comment]
### UPDATE 2023-02-22
**`rnsh` v0.0.12 is now required** to connect to the BBS. Run `pip3 install -U rnsh` to update. As part of the update, the destination hash changed. The old hash (beginning with `14`) will no longer work, use the hash `feda42292d882c5586e548d79b5af27b`.
